.github/workflows/pull_request.yml

```diff
@@ -31,13 +31,22 @@ defaults:
     shell: bash
 
 jobs:
+  os-prep:
+    runs-on: ubuntu-latest
+    outputs:
+      os: ${{ steps.set-os.outputs.operating-systems }}
+    steps:
+      - name: OS Preparation
+        id: set-os
+        uses: kiegroup/kogito-pipelines/.ci/actions/os-preparation@main
   build-chain:
+    needs: os-prep
     concurrency:
       group: pull_request-${{ github.event_name }}-${{ github.head_ref }}-${{ matrix.os }}-${{ matrix.java-version }}
       cancel-in-progress: true
     strategy:
       matrix:
-        os: [ ubuntu-latest, windows-latest ]
+        os: ${{ fromJSON(needs.os-prep.outputs.os) }}
```

Comment:
Why would we do it like this? We always want to run on Windows. What's the reason not to?

Reply:
It is using file from https://github.com/kiegroup/kogito-pipelines/blob/main/.github/matrix-os.json to determine the os to run on (so still ubuntu and windows)

Reply:
Then what's the `windows_check` label for?

Reply:
will be handled in https://github.com/kiegroup/kogito-pipelines/blob/main/.github/matrix-os.json#L12

Reply:
Hello @triceo , the idea behind this was to centralize the Operating Systems configuration for kogito projects in a single place, i.e., [https://github.com/kiegroup/kogito-pipelines/](https://github.com/kiegroup/kogito-pipelines/). For all those projects that use by default both windows and ubuntu won't need to add the `windows_check` label, but all the others yes (if needed). Another thing is that by centralizing this configuration, whenever there will be the need to update the OS config for pull requests, we won't have to do it on every project but only here https://github.com/kiegroup/kogito-pipelines/blob/main/.github/matrix-os.json